### PR TITLE
2460 Add kwargs for np.pad API

### LINKED ISSUE #2460: How do I to use a Pad transform with a constant value?
### Discussed in https://github.com/Project-MONAI/MONAI/discussions/2459

<div type='discussions-op-text'>

<sup>Originally posted by **vanossj** June 26, 2021</sup>
I'm looking at the various pad transforms that end up using `np.pad` such as `SpacialPad`. I can set the `mode="constant"` but how do I provide the constant value to use? `np.pad` has a `constant_values` param but I don't see that getting set in `SpacialPad`</div>

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2460 .

### Description
This transform added the `**kwargs` additional args for `np.pad` in all the padding transforms and added unit tests.
Also fixed a minor typo in doc-string.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by yiheng-wang-nv — APPROVED** [review]

File: monai/transforms/croppad/array.py [hunk 1]
```diff
@@ -58,28 +58,33 @@ class SpatialPad(Transform):
...
+            If its components have non-positive values, the corresponding size of input image will be used
+            (no padding). for example: if the spatial size of input data is [30, 30, 30] and
+            `spatial_size=[32, 25, -1]`, the spatial size of output data will be [32, 30, 30].
         method: {``"symmetric"``, ``"end"``}
             Pad image symmetric on every side or only pad at the end sides. Defaults to ``"symmetric"``.
         mode: {``"constant"``, ``"edge"``, ``"linear_ramp"``, ``"maximum"``, ``"mean"``,
             ``"median"``, ``"minimum"``, ``"reflect"``, ``"symmetric"``, ``"wrap"``, ``"empty"``}
             One of the listed string values or a user supplied function. Defaults to ``"constant"``.
             See also: https://numpy.org/doc/1.18/reference/generated/numpy.pad.html
+        kwargs: other args for `np.pad` API, note that `np.pad` treats channel dimension as the first dimension.
```
Comment (wyli) [inline monai/transforms/croppad/array.py:72]: could be `pad_kwargs` or `np_kwargs`

File: tests/test_border_pad.py [hunk 2]
```diff
@@ -51,6 +51,23 @@ def test_pad_shape(self, input_param, input_data, expected_val):
...
         self.assertAlmostEqual(result.shape, expected_val.shape)
 
+    def test_pad_kwargs(self):
+        padder = BorderPad(
+            spatial_border=2,
+            mode="constant",
+            constant_values=(
+                (
+                    0,
+                    0,
```
Comment (wyli) [inline tests/test_border_pad.py:61]:
```suggestion
                    0
```

File: monai/transforms/croppad/array.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/transforms/croppad/array.py:72]: OK, let me change it to `np_kwargs` as it's used for numpy API and this class is `SpatialPad`, no need to use the word `pad`.

Thanks.

Comment (Nic-Ma) [reply monai/transforms/croppad/array.py:72]: updated in the latest commit.
Thanks.

File: tests/test_border_pad.py (hunk 2 above)
Comment (Nic-Ma) [reply tests/test_border_pad.py]: updated in the latest commit.

Thanks.

Comment (Nic-Ma) [reply tests/test_border_pad.py]: Thanks for your fix, I misunderstood your comments, I thought you want to use `(0)` to replace `(0, 0)`..

**Pull request merged**